provider/github/watcher.go

```diff
@@ -169,6 +180,37 @@ func (w *Watcher) watchLoop(
 	}
 }
 
+// logGitHubErr logs an ErrGitHubAPI at debug or error level, depending on the
+// last time it was logged on error level
+func (w *Watcher) logGitHubErr(
```

Comment:
I don't like such cryptic signatures and var names :disappointed: 

I'd prefer something like:
```go
const errLogThrottlingInterval = 5 * time.Minute // because we're throttling

type lastErrLog struct {
	err  string
	when time.Time // instead of `last`
}

func (w *Watcher) logGitHubErr(
	ctx context.Context,
	history map[*lookout.RepositoryInfo]lastErrLog, // more meaningful
	repo *lookout.RepositoryInfo,
	resp *github.Response,
	err error,
	msg string, // we're logging a msg; this way there is no duplicate `err` in var names
) {
	lastRepoError := history[repo] // I find this way easier to remember

        var shouldBeLogged bool // I don't understand `asErr` meaning
	shouldBeLogged := lastRepoError.err != err.Error() ||
		time.Now().After(lastRepoError.when.Add(errLogThrottlingInterval))

        ...
```

But whatever you prefer :wink:

Reply:
:up: this is a friendly suggestion, not a blocker one ;)
Feel free to mark it as `resolved` if it does not suit you.

Reply:
:+1: I added most of your suggestions in da8b256.